compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -539,34 +540,41 @@ BOperatorSymbol createBuiltinMethodSymbol(BLangBuiltInMethod method, BType type,
         return new BOperatorSymbol(names.fromString(method.getName()), null, opType, null, opcode);
     }
 
-    BOperatorSymbol createTypeAssertionSymbol(BType type, BType retType) {
+    BOperatorSymbol createTypeCastSymbol(BType type, BType retType) {
         List<BType> paramTypes = Lists.of(type);
         BInvokableType opType = new BInvokableType(paramTypes, retType, null);
-        return new BOperatorSymbol(Names.ASSERTION_OP, null, opType, null, InstructionCodes.TYPE_ASSERTION);
+        return new BOperatorSymbol(Names.CAST_OP, null, opType, null, InstructionCodes.TYPE_CAST);
     }
 
-    BSymbol getExplicitlyTypedExpressionSymbol(BType sourceType, BType targetType) {
-        int sourceTypeTag = sourceType.tag;
-        if (types.isValueType(sourceType)) {
-            if (sourceType == targetType) {
-                return Symbols.createCastOperatorSymbol(sourceType, targetType, symTable.errorType, false, true, 
-                                                        InstructionCodes.NOP, null, null);
-            }
+    BSymbol getNumericConversionOrCastSymbol(BLangTypeConversionExpr conversionExpr, BType sourceType,
+                                             BType targetType) {
+        if (targetType.tag == TypeTags.UNION &&
+                ((BUnionType) targetType).memberTypes.stream()
```

Comment:
Better to refactor the second condition to a separate variable.

Reply:
Do you mean moving
```java
((BUnionType) targetType).memberTypes.stream()
                        .filter(memType -> types.isBasicNumericType(memType)).count() > 1
```

This depends on the first condition (`targetType.tag == TypeTags.UNION`) being true - since we are casting to union type.

Reply:
Yes, I meant the stream. Maybe we can split the 2nd conditions and move it inside the `if` condition and use a nested `if` block? I don't think we've used an stream result as an condition anywhere else (not 100% sure).

@SupunS - WDYT?